Program/23bca284-q10.ipynb

```diff
@@ -0,0 +1,316 @@
+   "source": [
+    "query =\"select cname, match, run, (run * 1.0 / match) as average from cricketer\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "3c7e3240",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<sqlite3.Cursor at 0x23ff23a61f0>"
+      ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "c.execute(query)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "655e179e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[('Player1', 10, 150, 15.0), ('Player2', 12, 180, 15.0), ('Player3', 8, 200, 25.0), ('Player4', 15, 130, 8.666666666666666), ('Player5', 11, 170, 15.454545454545455), ('Player6', 9, 160, 17.77777777777778), ('Player7', 14, 140, 10.0), ('Player8', 13, 190, 14.615384615384615)]\n"
+     ]
+    }
+   ],
+   "source": [
+    "rec=c.fetchall()\n",
+    "print(rec)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "8c998d4f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import csv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "id": "5087f54c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "players = [\n",
+    "    ('cid','cname','match','run','average'),\n",
+    "    (1, 'Player1', 10, 150,15.0),\n",
+    "    (2, 'Player2', 12, 180,15.0),\n",
+    "    (3, 'Player3', 8, 200,25.0),\n",
+    "    (4, 'Player4', 15, 130,8.666),\n",
+    "    (5, 'Player5', 11, 170,15.45),\n",
+    "    (6, 'Player6', 9, 160,17.77),\n",
+    "    (7, 'Player7', 14, 140,10.0),\n",
+    "    (8, 'Player8', 13, 190, 14.615)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "id": "7cf4bfe3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('D:/23BCA284/sqlite3/csv/player.csv','w') as f:\n",
+    "    w=csv.writer(f)\n",
+    "    w.writerows(players)  \n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "id": "a3fc0bf5",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "['cid', 'cname', 'match', 'run', 'average']\n",
+      "[]\n",
+      "['1', 'Player1', '10', '150', '15.0']\n",
+      "[]\n",
+      "['2', 'Player2', '12', '180', '15.0']\n",
+      "[]\n",
+      "['3', 'Player3', '8', '200', '25.0']\n",
+      "[]\n",
+      "['4', 'Player4', '15', '130', '8.666']\n",
+      "[]\n",
+      "['5', 'Player5', '11', '170', '15.45']\n",
+      "[]\n",
+      "['6', 'Player6', '9', '160', '17.77']\n",
+      "[]\n",
+      "['7', 'Player7', '14', '140', '10.0']\n",
+      "[]\n",
+      "['8', 'Player8', '13', '190', '14.615']\n",
+      "[]\n"
+     ]
+    }
+   ],
+   "source": [
+    "with open('D:/23BCA284/sqlite3/csv/player.csv','r') as f:\n",
```

Comment:
Any other way to open a file ?

Reply:
yes, we can open the file with use of open().